### PR TITLE
Xsd datetime duration

### PR DESCRIPTION
**Pull request by gkellogg**
Some updates to date time tests based on things I noticed in my implementation.

One thing not addressed, is how to handle `xsd:time("23:59:60")` as I don't see anything specific in F&O that addresses normalizing this value, unlike "24:00:00" which is specifically called out.

It doesn't seem that these tests were ever added to the main repo SEP-002 information, although I'm not sure there's a consensus on where tests belong, as the rdf-tests repo has also been mentioned.

### CODE REVIEW

File: tests/xsd_functions/construct_time-02.srx [hunk 1]
```diff
@@ -4,9 +4,15 @@
...
-		<result></result>
+		<result>
+			<binding name="time"><literal datatype="http://www.w3.org/2001/XMLSchema#time">00:00:01</literal></binding>
+		</result>
+		<result>
+			<binding name="time"><literal datatype="http://www.w3.org/2001/XMLSchema#time">05:60:00</literal></binding>
+		</result>
+		<result>
+			<binding name="time"><literal datatype="http://www.w3.org/2001/XMLSchema#time">00:00:61</literal></binding>
+		</result>
```
Comment (kasei) [inline tests/xsd_functions/construct_time-02.srx]: I'm not sure about these. According to the [lexical forms of the time components](https://www.w3.org/TR/xmlschema11-2/#nt-seFrag):

```
[59]   hourFrag ::= ([01] digit) | ('2' [0-3])
[60]   minuteFrag ::= [0-5] digit
[61]   secondFrag ::= ([0-5] digit) ('.' digit+)?
```

How does that square with successfully casting these strings to the xsd:time constructor?

```
"24:00:01"
"05:60:00"
"00:00:61"
```

File: tests/xsd_functions/construct_date-02.srx [hunk 2]
```diff
@@ -4,9 +4,9 @@
...
 </head>
 <results>
-		<result></result>
-		<result></result>
-		<result></result>
-		<result></result>
+		<result><binding name="date"><literal datatype="http://www.w3.org/2001/XMLSchema#date">2000-00-01</literal></binding></result>
+		<result><binding name="date"><literal datatype="http://www.w3.org/2001/XMLSchema#date">2000-13-01</literal></binding></result>
+		<result><binding name="date"><literal datatype="http://www.w3.org/2001/XMLSchema#date">2000-06-00</literal></binding></result>
+		<result><binding name="date"><literal datatype="http://www.w3.org/2001/XMLSchema#date">2000-06-32</literal></binding></result>
```
Comment (kasei) [inline tests/xsd_functions/construct_date-02.srx]: The [lexical form for date components](https://www.w3.org/TR/xmlschema11-2/#nt-yrFrag) says:

```
[57]   monthFrag ::= ('0' [1-9]) | ('1' [0-2])
[58]   dayFrag ::= ('0' [1-9]) | ([12] digit) | ('3' [01])
```

I'm not sure how we could expect to get these xsd:date values out of the constructor for these string values:

```
"2000-00-01"
"2000-13-01"
"2000-06-00"
"2000-06-32"
```

**Review by kasei — CHANGES_REQUESTED** [review]
Sorry it took so long to look into these changes.

File: tests/xsd_functions/construct_time-02.srx (hunk 1 above)
Comment (kasei) [reply tests/xsd_functions/construct_time-02.srx]: Today I learned that:

> [Leap seconds are not supported by the types defined here.](https://www.w3.org/TR/xmlschema11-2/#d-t-values)

But also realized I missed the [lexical inclusion of "24:00:00"](https://www.w3.org/TR/xmlschema11-2/#nt-eodFrag). So I still think that none of the above are valid lexical forms.

Comment (gkellogg) [reply tests/xsd_functions/construct_time-02.srx]: Agreed, these should be invalid. How should a processor behave when it sees these? Is the correct output a results with no bindings, as you had it before?

File: tests/xsd_functions/construct_date-02.srx (hunk 2 above)
Comment (gkellogg) [reply tests/xsd_functions/construct_date-02.srx]: Agreed, in retrospect, given the syntactic limitations.

**Comment by gkellogg** [comment]
@kasei is simply reverting changes to construct-date.srx and construct-time.srx what you propose? Or, is there some variation on acceptable output that is needed?

File: tests/xsd_functions/construct_time-02.srx (hunk 1 above)
Comment (kasei) [reply tests/xsd_functions/construct_time-02.srx]: I believe the no-bindings output is correct, yes.

**Comment by kasei** [comment]
> @kasei is simply reverting changes to construct-date.srx and construct-time.srx what you propose? Or, is there some variation on acceptable output that is needed?

I think reverting is probably best. The corresponding `-01` tests cover valid cases (though as I write this, I'm noticing that one of the tests in `construct_time-01.rq` also uses a lexical form with 60 seconds, so that probably needs correcting.

**Comment by gkellogg** [comment]
I reverted construct_date-02.srx and construct_time-02.srx. Any problem with construct_time-01.rq wasn't introduced in this PR. We could change it to `23:59:59`, but I don't know what that adds, unless we leave it, and show an empty result in the .srx file.

**Review by kasei — APPROVED** [review]

**Pull request merged**